pkg/mermaid/README.md

```diff
@@ -0,0 +1,47 @@
+**Mermaid diagram generation design doc**
+
+**Objective**
+
+- Explore alternatives for PlantUML for diagram generation.
+
+**Background**
+
+Generating diagram using PlantUML requires making HTTP requests to the PlantUML server. In order to support systems generating several diagrams at once, it is not advisable to rely on an external service which could also affect performance and reliability.
```

Comment:
I think some detailing around the ability to generate diagrams as part of CI is hindered when needing an external Plantuml dependency?